mne/viz/epochs.py

```diff
@@ -536,3 +724,172 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, proj=False, n_fft=256,
     if show:
         plt.show()
     return fig
+
+
+def _plot_traces(params):
+    """ Helper for plotting concatenated epochs """
+    n_channels = params['n_channels']
+    types = params['types']
+    lines = params['lines']
+    data = params['data']
+    offsets = params['offsets']
+
+    tick_list = []
+    start_idx = int(params['t_start'] / len(params['epochs'].times))
+    labels = params['labels'][start_idx:]
+    params['ax'].set_xticklabels(labels)
+    # do the plotting
+    for ii in range(n_channels):
+        ch_ind = ii + params['ch_start']
+        if ii >= len(lines):
+            break
+        elif ch_ind < len(params['picks']):
+            # scale to fit
+            ch_name = params['ch_names'][ch_ind]
+            tick_list += [ch_name]
+            offset = offsets[ii]
+            end = params['t_start'] + params['duration']
+            this_data = data[ch_ind][params['t_start']:end]
+
+            # subtraction here gets correct orientation for flipped ylim
+            ydata = offset - this_data
+            xdata = params['times'][:params['duration']]
+            segments = np.split(np.array(zip(xdata, ydata)),
+                                params['n_epochs'])
+
+            lines[ii].set_segments(segments)
+            vars(lines[ii])['ch_name'] = ch_name
+            color = params['color'][types[ch_ind]]
+            vars(lines[ii])['def_color'] = params['color'][types[ch_ind]]
+            lines[ii].set_color(color)
+        else:
+            # "remove" lines
+            lines[ii].set_xdata([])
+            lines[ii].set_ydata([])
+    # deal with event lines
+    if params['epoch_times'] is not None:
+        # find events in the time window
+        epoch_times = params['epoch_times']
+        event_line = params['event_line']
+        xs = list()
+        ys = list()
+        for tt in epoch_times:
+            xs += [tt, tt, np.nan]
+            ylim = params['ax'].get_ylim()
+            ys += [ylim[0], ylim[1], np.nan]
+        event_line.set_xdata(xs)
+        event_line.set_ydata(ys)
+
+    # finalize plot
+    params['ax'].set_xlim(params['times'][0],
+                          params['times'][0] + params['duration'], False)
+    params['ax'].set_yticklabels(tick_list)
+    params['vsel_patch'].set_y(params['ch_start'])
+    params['fig'].canvas.draw()
+
+
+def _plot_onscroll(event, params):
+    """
+    Scroll events.
```

Comment:
nitpick but it'll be a pain to update everything later if you don't stick to the conventions: https://www.python.org/dev/peps/pep-0257/#one-line-docstrings. It should all be in one line